### ISSUE TITLE
qwen-image-edit-plus bug

### ISSUE THREAD

**Issue by zhangquanwei962** [PKU-YuanGroup/Edit-R1#15]
你好，我发现训练过程貌似有问题，我把https://github.com/PKU-YuanGroup/Edit-R1/blob/f86ef78c4656e4f5410e8defaf56ec23c68e8294/scripts/train_nft_qwen_image_edit.py#L827
改成了
```
            generated_images = images.cpu()
            ref_images_tensor = torch.stack([TF.to_tensor(img) for img in ref_images]).cpu()

            # Concatenate horizontally (ref | generated)
            comparison_images = torch.cat([ref_images_tensor, generated_images], dim=-1)  # width dimension
            images_to_log = comparison_images```

```

这样子之后，为了看原图和编辑生成之后的图是否对应，结果发现训练log的图不对应。。。

<img width="380" height="376" alt="Image" src="https://github.com/user-attachments/assets/4a9e4eb1-708a-4aca-a907-bc8a862b4417" />

<img width="379" height="376" alt="Image" src="https://github.com/user-attachments/assets/5be49c44-cf0b-4cab-aed1-502aa91d13fc" />

<img width="1197" height="246" alt="Image" src="https://github.com/user-attachments/assets/3d13a2ff-3858-4be0-9d71-25f12ea1ffe8" />看起来这个每个GPU的3个样本prompt不一样，但貌似没有prompt和图片对应？
8卡执行，没有做其他的改动

**Comment by zhangquanwei962** [comment]
> 你好，我发现训练过程貌似有问题，我把
> 
> [Edit-R1/scripts/train_nft_qwen_image_edit.py](https://github.com/PKU-YuanGroup/Edit-R1/blob/f86ef78c4656e4f5410e8defaf56ec23c68e8294/scripts/train_nft_qwen_image_edit.py#L827)
> 
> Line 827 in [f86ef78](/PKU-YuanGroup/Edit-R1/commit/f86ef78c4656e4f5410e8defaf56ec23c68e8294)
> 
>  images_to_log = images.cpu()  # from last sampling batch on this rank 
> 
> 改成了
> ```
>             generated_images = images.cpu()
>             ref_images_tensor = torch.stack([TF.to_tensor(img) for img in ref_images]).cpu()
> 
>             # Concatenate horizontally (ref | generated)
>             comparison_images = torch.cat([ref_images_tensor, generated_images], dim=-1)  # width dimension
>             images_to_log = comparison_images```
> ```
> 
> 这样子之后，为了看原图和编辑生成之后的图是否对应，结果发现训练log的图不对应。。。
> 


我已经解决了这个bug，对于qwen-image-edit-2509 作者开源的代码没有针对per gpu batchsize=3做改变，导致每次都是多图融合，需要我提个PR吗？
